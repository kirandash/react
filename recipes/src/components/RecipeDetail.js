import React from 'react';

const RecipeDetail = (props) => {
    if(!props.recipe){
        return (
            <p style={props.style}>Please select a recipe to see the detail!</p>
        )
    }
    return (
        <div style={props.style}>
            <h2>{props.recipe.name}</h2>
            <img src={props.recipe.image} width="200"/>
            <div>
                <span>{props.recipe.category}</span>
                <span>{props.recipe.calories}</span>
            </div>
            <h3>Ingredients:</h3>
            <ul>
                {props.recipe.ingredients.map(ingredient => (
                    <li key={ingredient}>
                        {ingredient}
                    </li>
                ))}
            </ul>
            <h3>Steps:</h3>
            <ol>
                <li>First do this</li>
                <li>Second do this</li>
                <li>Third do this</li>
            </ol>
        </div>
    )
};

export default RecipeDetail;