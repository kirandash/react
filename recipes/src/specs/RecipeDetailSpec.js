import renderer from 'react-test-renderer';
import RecipeDetail from '../components/RecipeDetail';
import { BrowserRouter } from 'react-router-dom'; // Since RecipeDetail has a Link. So it must be wrapped in BrowserRouter to avoid any error
import React from 'react';

describe('<RecipeDetail />', () => {
    let testRecipe;

    beforeEach(() => {
        testRecipe = {
            id: 1,
            name: 'Test recipe',
            category: 'Test category',
            ingredients: ['Ing 1', 'Ing 2'],
            steps: ['Step 1', 'Step 2']
        }; // Since we are modifying testRecipe variable in diff test cases. It's better to define it before Each test case runs
    });

    test('Should not break if no recipe passed', () => {
        const component = renderer.create(
            <BrowserRouter>
                <RecipeDetail />
            </BrowserRouter>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('Should render correctly a recipe', () => {
        const component = renderer.create(
            <BrowserRouter>
                <RecipeDetail recipe={testRecipe} />
            </BrowserRouter>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('Should correctly assign a class name', () => {
        const component = renderer.create(
            <BrowserRouter>
                <RecipeDetail recipe={testRecipe} className="classname-test" />
            </BrowserRouter>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('Should render recipe w/o ingredients', () => {
        delete testRecipe.ingredients;
        const component = renderer.create(
            <BrowserRouter>
                <RecipeDetail recipe={testRecipe} />
            </BrowserRouter>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
    test('Should render recipe w/o steps', () => {
        delete testRecipe.steps;
        const component = renderer.create(
            <BrowserRouter>
                <RecipeDetail recipe={testRecipe} />
            </BrowserRouter>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
})