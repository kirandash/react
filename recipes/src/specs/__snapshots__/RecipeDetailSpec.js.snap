// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecipeDetail /> Should correctly assign a class name 1`] = `
<div
  className="p2 bg-white classname-test"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients:
  </h3>
  <ul>
    <li>
      Ing 1
    </li>
    <li>
      Ing 2
    </li>
  </ul>
  <h3>
    Steps:
  </h3>
  <ol>
    <li>
      Step 1
    </li>
    <li>
      Step 2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See more
  </a>
</div>
`;

exports[`<RecipeDetail /> Should not break if no recipe passed 1`] = `
<p
  className="h3 p2 bg-white italic center"
>
  Please select a recipe to see the detail!
</p>
`;

exports[`<RecipeDetail /> Should render correctly a recipe 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients:
  </h3>
  <ul>
    <li>
      Ing 1
    </li>
    <li>
      Ing 2
    </li>
  </ul>
  <h3>
    Steps:
  </h3>
  <ol>
    <li>
      Step 1
    </li>
    <li>
      Step 2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See more
  </a>
</div>
`;

exports[`<RecipeDetail /> Should render recipe w/o ingredients 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients:
  </h3>
  <h3>
    Steps:
  </h3>
  <ol>
    <li>
      Step 1
    </li>
    <li>
      Step 2
    </li>
  </ol>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See more
  </a>
</div>
`;

exports[`<RecipeDetail /> Should render recipe w/o steps 1`] = `
<div
  className="p2 bg-white"
>
  <h2
    className="h2"
  >
    Test recipe
  </h2>
  <img
    alt="Test recipe"
    className="fit"
  />
  <div>
    <span>
      Test category
    </span>
    <span />
  </div>
  <h3>
    Ingredients:
  </h3>
  <ul>
    <li>
      Ing 1
    </li>
    <li>
      Ing 2
    </li>
  </ul>
  <h3>
    Steps:
  </h3>
  <a
    href="/recipe/1"
    onClick={[Function]}
  >
    See more
  </a>
</div>
`;
